### DISCUSSION TITLE
Documentation to create an artillery engine?

### DISCUSSION THREAD

**Discussion by thelukelynch** [artilleryio/artillery#1593] [Artillery]
Hi All,

New person to artillery here. I was looking for a valid plugin to enable ntlm auth for artillery requests. Current solutions that I've found haven't been updated in years. I was hoping to maybe create a new one to solve my issue but I can't seem to find a good spot to start

Thanks

**Comment by hassy** [comment]
hi @DrWalrus1, you could look at these engines:

https://github.com/artilleryio/artillery-engine-kinesis

https://github.com/artilleryio/artillery-engine-playwright

Or the built-in WebSocket engine as it's quite lightweight.

The engine APIs have been maintained in a backwards-compatible manner for a few years now, so it should not matter if a specific engine you were looking at hasn't been updated in a while.

**Reply by thelukelynch** [reply]
Thanks for your response.

I'll take a look at those engines.

The engine I was looking at that was struggling to run was [artillery-engine-http-ntml](https://www.npmjs.com/package/artillery-engine-http-ntml). Seems to have a lot of issues.

**Reply by hassy** [reply]
I'm not familiar with NTLM but if it's an authentication method, would it perhaps make sense for the functionality to be a plugin, rather than an entire new engine?

A plugin can add functionality to an existing engine, e.g. see https://github.com/artilleryio/artillery-plugin-metrics-by-endpoint. Whereas an engine provides a way to run an entire scenario definition.
